src/core/src/CoreMain.py

```diff
@@ -59,6 +60,15 @@ def __init__(self, argv):
             telemetry_writer.set_task_name(Constants.TelemetryTaskName.AUTO_ASSESSMENT if execution_config.exec_auto_assess_only else Constants.TelemetryTaskName.EXEC)
             patch_operation_requested = execution_config.operation.lower()
 
+            # clean up temp folder before any operation execution begins from Core
+            if bootstrapper.env_layer is not None \
+                    and execution_config is not None \
+                    and execution_config.temp_folder is not None \
```

Comment:
When would any of these three conditions evaluate to false?

Reply:
bootstrapper.env_layer is not None \
                    and execution_config is not None \ 

this is just a defensive check, in case they are not instantiated

[execution_config.temp_folder is not None] can happen is handler does not send temp_folder

Reply:
None of the conditions _should_ evaluate to false here. We can deterministically guarantee that with correct coding elsewhere so can be removed.

It's more understandable in the second location where you have this because of the various execution paths the code can take before hitting that.

Reply:
Resolved